### ISSUE TITLE
When using assumed roles, you can't attribute audit event actions to a specific aws user.

### ISSUE THREAD

**Issue by tjsingleton** [kubernetes-sigs/aws-iam-authenticator#263]
TL;DR: I don't think it's possible to know for certain who the actor is in the kubernetes audit logs when the caller assumed a role.

The goal is that I can trace an k8s audit record to a single aws user who made the api call. We manage our users in a central account and they assume a shared role into the accounts where the actual aws resources live. For events in cloudtrail aws logs the accessKeyId of the role assumption session. Therefore I can correlate that  accessKeyId with the original "log in". (Described here: https://aws.amazon.com/blogs/security/how-to-audit-cross-account-roles-using-aws-cloudtrail-and-amazon-cloudwatch-events/) 

However, the k8s audit logs do not include the accessKeyId. 

Example:

```
...
    "user": {
        "username": "platform:john.doe-617090408934",
        "uid": "heptio-authenticator-aws:617090408934:PBNMABOXMHFFMNIGGNZGX",
        "groups": [
            "system:masters",
            "system:authenticated"
        ]
    },
...
```

The uid includes the id of the assumed role, but that isn't unique to the session and doesn't identify who called it. The username field includes the role and the session name, however the session name is generated client side and could be forged. We've included the user's name by convention, but it can't be trusted. (The docs for the iam-authenticator call out that this can't be trusted https://github.com/kubernetes-sigs/aws-iam-authenticator#note-for-federated-users)

If the accessKeyId was available as part of how iam-authenticator generates the username or uid for the audit event, you could correlate it back to the cloudtrail entry.

**Comment by jritchey** [comment]
@tjsingleton in your example is `PBNMABOXMHFFMNIGGNZGX` the accessKeyId you are looking for? If you search CloudTrail and filter on `Resource Name` and `PBNMABOXMHFFMNIGGNZGX` does it lead back to the IAM user you expected?

**Comment by tjsingleton** [comment]
Yeah, it's the assumed role id.
